Update equalIgnoringASCIICase() to take in 2 spans, instead of a raw pointer and a span

cdumez opened the pull request:
#### 56ee9c85d0f254dc1515c1f6356a7dff89995c52
<pre>
Update equalIgnoringASCIICase() to take in 2 spans, instead of a raw pointer and a span
<a href="https://bugs.webkit.org/show_bug.cgi?id=273835">https://bugs.webkit.org/show_bug.cgi?id=273835</a>

Reviewed by Darin Adler.

* Source/WTF/wtf/text/StringCommon.h:
* Source/WTF/wtf/text/StringView.h:
(WTF::startsWithIgnoringASCIICase):
(WTF::endsWithIgnoringASCIICase):

Canonical link: <a href="https://commits.webkit.org/278550@main">https://commits.webkit.org/278550@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b93f4e95b4e021e8a8409209547be534f00a2cb5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50917 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30216 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3239 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54175 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1607 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36492 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1274 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41465 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53015 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27861 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43863 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22595 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25183 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/1120 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/9357 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/44252 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/47168 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1190 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55769 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/50419 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26021 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/1090 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48876 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27276 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43936 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47959 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28149 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/57894 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7381 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27007 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11918 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/text/StringView.h:
```diff
@@ -1250,13 +1250,13 @@ inline size_t findIgnoringASCIICase(StringView source, StringView stringToFind,
-            return findIgnoringASCIICase(source.span8().first(searchLength), stringToFind.span8(), static_cast<size_t>(start));
-        return findIgnoringASCIICase(source.span8().first(searchLength), stringToFind.span16(), static_cast<size_t>(start));
+            return findIgnoringASCIICase(source.span8(), stringToFind.span8(), static_cast<size_t>(start));
+        return findIgnoringASCIICase(source.span8(), stringToFind.span16(), static_cast<size_t>(start));
```
I don't understand the purpose of these two calls to `static_cast`. Should just pass `start`.

cdumez replied:
I tried dropping them but then the build failed. The issue is that the function call becomes ambiguous.

darinadler commented on Source/WTF/wtf/text/StringView.h:
```diff
@@ -1250,13 +1250,13 @@ inline size_t findIgnoringASCIICase(StringView source, StringView stringToFind,
-        return findIgnoringASCIICase(source.span16().first(searchLength), stringToFind.span8(), static_cast<size_t>(start));
-    return findIgnoringASCIICase(source.span16().first(searchLength), stringToFind.span16(), static_cast<size_t>(start));
+        return findIgnoringASCIICase(source.span16(), stringToFind.span8(), static_cast<size_t>(start));
+    return findIgnoringASCIICase(source.span16(), stringToFind.span16(), static_cast<size_t>(start));
```
I don't understand the purpose of these two calls to `static_cast`. Should just pass `start`.

cdumez committed:
Update equalIgnoringASCIICase() to take in 2 spans, instead of a raw pointer and a span
https://bugs.webkit.org/show_bug.cgi?id=273835

Reviewed by Darin Adler.

* Source/WTF/wtf/text/StringCommon.h:
* Source/WTF/wtf/text/StringView.h:
(WTF::startsWithIgnoringASCIICase):
(WTF::endsWithIgnoringASCIICase):

Canonical link: https://commits.webkit.org/278550@main
